dimod/sampleset.py

```diff
@@ -453,6 +453,27 @@ def infer_vartype(samples_like):
 
     raise ValueError("given samples_like is of an unknown vartype")
 
+def drop_variables(sampleset, variables_to_drop):
+    """Drop variables from a sampleset
```

Comment:
We should be clear we are returning _a new_ sampleset (a copy), not modifying the original one.